### PR TITLE
fix(templates/express): dynamically import `chokidar` so it doesn't crash production

### PR DESCRIPTION
**Pull request by mcansh**
prior to this change if you built the remix app, pruned dev only dependencies and attempted to `npm start` the server would bail as it tried to import chokidar

Closes: #

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

**Review by MichaelDeBoey — APPROVED** [review]
I think we should merge this into `main` + we should probably also update dev_v2 docs with this

**Comment by ngbrown** [comment]
This was also solved in https://github.com/remix-run/remix/pull/6922#pullrequestreview-1546618836, along with making CJS easy to switch to and fixing a race condition.

**Comment by brophdawg11** [comment]
As a template change I agree this should go to `main` - but we should hold off until v2 releases since I think the express server is pretty different in `main` (v1 dev server) and `dev` (v2 dev server).

**Comment by mcansh** [comment]
rebased!

i originally opened this PR before v2 went out which is the reason for it being pointed to `dev`, not really in the mood to go through the whole `git rebase --onto main dev (git branch --show-current) --committer-date-is-author-date` shooting match to keep this PR but repoint it to `main` so if that's desired instead of merging + cherrypicking this one, i can close and bring it over to a new PR pointed to `main`

**Review by pcattori — APPROVED** [review]

**Pull request merged**
